lightning-persister/src/util.rs

```diff
@@ -39,21 +32,23 @@ fn path_to_windows_str<T: AsRef<OsStr>>(path: T) -> Vec<winapi::shared::ntdef::W
 }
 
 #[allow(bare_trait_objects)]
-pub(crate) fn write_to_file<D: DiskWriteable>(path: PathBuf, filename: String, data: &D) -> std::io::Result<()> {
-	fs::create_dir_all(path.clone())?;
+pub(crate) fn write_to_file<W: Writeable>(filename_with_path: String, data: &W) -> std::io::Result<()> {
+	let mut path_parts: Vec<&str> = filename_with_path.split("/").collect();
```

Comment:
Can we right-split and not collect into a Vec?

Reply:
hm well if we split with char '/' it returns DoubleEndedIterator so don't need rsplit.
but if we're not okay with assuming '/' as separator then we can't do this anyway?

Reply:
tried doing this by folding the iterator into a String but I'm not really sure if it's better than just collecting and joining

Reply:
oh looks like you can just collect straight into a PathBuf (which is what we want anyway) and avoids using the '/'.  Solves both problems at once? at least for here

Reply:
er I guess we're still splitting on "/" so doesn't entirely avoid it.  

I'm not really sure what to do with this PathBuf <--> String issue

Reply:
it's annoying because on both sides of the trait we want PathBuf but the trait is defined over a String key.

Reply:
ended up keeping everything String's and building paths manually format!() and std::path::MAIN_SEPARATOR.  Not sure if this is the best way though.

Finally convert to PathBuf inside of util::write_to_file when needed.